examples/read_write/read_write.c

```diff
@@ -85,8 +84,11 @@ int main(int argc, char *argv[]) {
 			printf("%02x ", buffer[i]);
 		printf("\n");
 	} else {
-		ret = gattlib_write_char_by_uuid(connection, &g_uuid, buffer, sizeof(buffer));
+      buffer[0] = (uint8_t) value_data;
+      buffer[1] = '\0';
+		ret = gattlib_write_char_by_uuid(connection, &g_uuid,buffer,1);
```

Comment:
`value_data` is initialized with `value_data = strtol(argv[4], NULL, 0);` and `strtol` returns a `long int`. Would not be the correct code simply being:

```
ret = gattlib_write_char_by_uuid(connection, &g_uuid, value_data, sizeof(value_data));
```

Reply:
Note First: I am not suggesting that my code is suitable to be merged.

The type of the data should be the same as the type of the characteristic at the remote peripheral "server" end, shouldnt it? In my remote "server" end , the characteristic is an 8 bit type. Bluez seems to know this since passing more than 2 bytes into bluez in my application is silently ignored and the value is not changed. (Note it is some time since I worked on it so I will have to go back and check the behaviour and get back to you) 

Bluez seems to want the type to be presented as an array of some 8 bit type that can be converted back at the remote end.